lib/Doctrine/DBAL/Migrations/Configuration/Connection/Loader/ConnectionHelperLoader.php

```diff
@@ -9,21 +9,17 @@
 
 class ConnectionHelperLoader implements ConnectionLoaderInterface
 {
-    /**
-     * @var string
-     */
+    /** @var string */
     private $helperName;
 
     /** @var  HelperSet */
     private $helperSet;
 
 
     /**
-     * ConnectionHelperLoader constructor.
-     * @param HelperSet $helperSet
      * @param string $helperName
      */
-    public function __construct(HelperSet $helperSet = null, $helperName)
+    public function __construct(?HelperSet $helperSet = null, $helperName)
```

Comment:
The signature needs updating: can't have an argument without default value after one with default value. I suggest removing the `null` default for `$helperSet`.

Reply:
Actually, you can.
![](https://scontent.fprg2-1.fna.fbcdn.net/v/t1.0-9/10367152_262849817251689_6553281229493693166_n.png?oh=0f97511c4b842eac4774cbbc1a3c153e&oe=5B322FFA)

Reply:
Yes and no: having a method signature like `foo($foo = 'foo', $bar)` will always require you to give all arguments, making the default value for `$foo` obsolete: https://3v4l.org/l1qrv

Since nobody could've been relying on the default value, we should drop it, especially since it was just there to allow for a nullable object type.